manual/src/cmds/tail-mod-cons.etex

```diff
@@ -150,6 +158,32 @@ tail-mod-cons are as follows:
   using a convenient monadic notation.
 \end{itemize}
 
+\paragraph{Note: OCaml call stack size} In OCaml 4.x and earlier,
+bytecode programs respect the "stack_limit" runtime parameter
+configuration (as set using "Gc.set" in the example above), or the "l"
+setting of the "OCAMLRUNPARAM" variable. Native programs ignore these
+settings and only respect the operating system native stack limit, as
+set by "ulimit" on Unix systems. Most operating systems run with
+a relatively low stack size limit by default, so stack overflow on
+non-tail-recursive functions are a common programming bug.
+
+Starting from OCaml 5.0, native code does not use the native system
+stack for OCaml function calls anymore, so it is not affected by the
+operating system native stack size; both native and bytecode programs
+respect the OCaml runtime's own limit. The runtime limit is set to
+a much higher default than most operating system native stacks, with
+a limit of at least 512MiB, so stack overflow should be much less
+common in practice. There is still a stack limit by default, as it
+remains useful to quickly catch bugs with looping non-tail-recursive
+functions. Without a stack limit, one has to wait for the whole memory
+to be consumed by the stack for the program to crash, which can take
+a long time and make the system unresponsive.
+
+This means that the "tail modulo constructor" transformation is less
+important on OCaml 5: it does improve performance noticeably in some
+cases, but it is not necessary for basic correctness for most
+use-cases.
```

Comment:
I don't completely agree with this remark on correctness. Certainly, this is true for internal computations of a program where the program can know in advance the required stack size. However, this is not the case for a library that cannot predict all its future usages nor for any computations driven by user inputs for a program.

Reply:
My experience is rather that most authors of a concrete program can make a guess on whether "hundreds of millions" is a common user-input size for their programs (very often the answer is "no"), and that most library authors authors do not worry too much about running in constant stack usage unless they start seeing user reports that something breaks on large inputs.

I might be a bit over-critical in discussing the usefulness of a feature I participated to implement, but I guess that this is better than being over-positive.